docs/rules/arrow-parens.md

```diff
@@ -177,6 +178,7 @@ a.then(foo => { if (true) {} });
 (a = 10) => a;
 ([a, b]) => a;
 ({a, b}) => a;
+const f = /** @type {number} */a => a + a;
```

Comment:
I think these examples don't reflect this change. We should add some correct examples with parens and comment inside.